chemprop/models/mpn.py

```diff
@@ -156,10 +138,20 @@ def __init__(self,
         super(MPN, self).__init__()
         self.atom_fdim = atom_fdim or get_atom_fdim()
         self.bond_fdim = bond_fdim or get_bond_fdim(atom_messages=args.atom_messages)
-        self.encoder = MPNEncoder(args, self.atom_fdim, self.bond_fdim)
+
+        self.features_only = args.features_only
+        self.use_input_features = args.use_input_features
+        self.device = args.device
+
+        if self.features_only:
+            return
+
+        #self.encoder = [MPNEncoder(args, self.atom_fdim, self.bond_fdim) for i in range(args.number_of_molecules)]
+        self.encoder = nn.ModuleList([MPNEncoder(args, self.atom_fdim, self.bond_fdim)
+                                      for i in range(args.number_of_molecules)])
```

Comment:
If you add the `shared_mpn` option to args, I _think_ this will work because it will create multiple pointers to the same model rather than creating separate models, but you should definitely test it:

```suggestion
        if args.shared_mpn:
            self.encoder = nn.ModuleList([MPNEncoder(args, self.atom_fdim, self.bond_fdim)] * args.number_of_molecules)
        else:
            self.encoder = nn.ModuleList([MPNEncoder(args, self.atom_fdim, self.bond_fdim)
                                          for _ in range(args.number_of_molecules)])
```

Also note that I like to use `_` instead of `i` in for loops when the index isn't actually used to show that fact.